module/core/src/Infrastructure/Handler/UpdateLanguageCommandHandler.php

```diff
@@ -21,19 +23,51 @@ class UpdateLanguageCommandHandler
      */
     private $repository;
 
+    /**
+     * @var LanguageQueryInterface
+     */
+    private $query;
+
     /**
      * @param LanguageRepositoryInterface $repository
+     * @param LanguageQueryInterface      $query
      */
-    public function __construct(LanguageRepositoryInterface $repository)
+    public function __construct(LanguageRepositoryInterface $repository, LanguageQueryInterface $query)
     {
         $this->repository = $repository;
+        $this->query = $query;
     }
 
     /**
      * @param UpdateLanguageCommand $command
      */
     public function __invoke(UpdateLanguageCommand $command)
     {
-        $this->repository->save($command->getCode(), $command->isActive());
+        $activeLanguages = $command->getLanguages();
+        $allLanguages = $this->query->getAll();
+        foreach ($allLanguages as $language) {
+            if ($this->hasCode($language, $activeLanguages)) {
+                $this->repository->save($language, true);
+            } else {
+                $this->repository->save($language, false);
+            }
```

Comment:
This can be done with simpler code.  
``
$hasCode = $this->hasCode($language, $activeLanguages);

$this->repository->save($language, $hasCode);
``